Auctions/MapToRedis.fs

```diff
@@ -3,146 +3,121 @@ open Auctions.Domain
 
 open StackExchange.Redis
 open System
+open FSharpPlus
+open FSharpPlus.Data
 
-let redisKey (str : string) = RedisKey.op_Implicit (str)
-let redisValueStr (str : string) = RedisValue.op_Implicit str
-let redisValueInt64 (v : int64) = RedisValue.op_Implicit v
-let redisValueFloat (v : float) = RedisValue.op_Implicit v
-let valueToKey (v : RedisValue) : RedisKey = RedisKey.op_Implicit (string v)
-let hashEntryStr key value = HashEntry(redisValueStr key, redisValueStr value)
-let hashEntryInt64 key value = HashEntry(redisValueStr key, redisValueInt64 value)
-let hashEntryFloat key value = HashEntry(redisValueStr key, redisValueFloat value)
+module Redis=
+  
+  // Codec definition From Fleece:
 
-let mapToHashEntries command = 
+  /// Encodes a value of a generic type 't into a value of raw type 'S.
+  type Encoder<'S, 't> = 't -> 'S 
+
+  /// Decodes a value of raw type 'S into a value of generic type 't, possibly returning an error.
+  type Decoder<'S, 't> = 'S -> Result<'t,string>
+
+  /// A decoder from raw type 'S1 and encoder to raw type 'S2 for string types 't1 and 't2.
+  type Codec<'S1, 'S2, 't1, 't2> = Decoder<'S1, 't1> * Encoder<'S2, 't2>
+
+  /// A decoder from raw type 'S1 and encoder to raw type 'S2 for type 't.
+  type Codec<'S1, 'S2, 't> = Codec<'S1, 'S2, 't, 't>
+
+  /// A codec for raw type 'S decoding to strong type 't1 and encoding to strong type 't2.
+  type SplitCodec<'S, 't1, 't2> = Codec<'S, 'S, 't1, 't2>
+
+  /// A codec for raw type 'S to strong type 't.
+  type Codec<'S, 't> = Codec<'S, 'S, 't>
+
+  let decode (d: Decoder<'i, 'a>) (i: 'i) : Result<'a,_> = d i
+  let encode (e: Encoder<'o, 'a>) (a: 'a) : 'o = e a
+
+  /// <summary>Initialize the field mappings.</summary>
+  /// <param name="f">An object initializer as a curried function.</param>
+  /// <returns>The resulting object codec.</returns>
+  let mapping f = (fun _ -> Ok f), (fun _ -> [])
+  let inline get fromRedis (o: HashEntry list) (key:string) =
+    let k = RedisValue.op_Implicit key
+    match o |> List.tryFind (fun p -> p.Name.Equals(k) ) with
+    | Some v-> fromRedis v.Value
+    | None -> failwithf "Could not find %s" key
+
+  let diApply combiner toBC (remainderFields: SplitCodec<'S, 'f ->'r, 'T>) (currentField: Codec<'S, 'f>) = //NOTE: From Fleece
+      (
+          Compose.run (Compose (fst remainderFields: Decoder<'S, 'f -> 'r>) <*> Compose (fst currentField)),
+          toBC >> (encode (snd currentField) *** encode (snd remainderFields)) >> combiner
+      )
+
+  /// <summary>Appends a field mapping to the codec.</summary>
+  /// <param name="fieldName">A string that will be used as key to the field.</param>
+  /// <param name="getter">The field getter function.</param>
```

Comment:
`fromRedis` and `toRedis` params are missing in this xmldocs

Reply:
I've added them 👍